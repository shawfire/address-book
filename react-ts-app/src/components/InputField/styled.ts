import styled from '@emotion/styled';
import { theme } from '../../theme';

export const StyledFieldWrapper = styled.div<{
  errorMessage: string;
}>(({ errorMessage }) => ({
  display: 'flex',
  flexDirection: 'column',
  '@media (max-width: 800px)': {
    margin: 'auto 16px',
    maxWidth: '420px'
  },
  input: {
    ...theme.input,
    height: '48px',
    borderRadius: '4px',
    paddingLeft: '16px',
    lineHeight: '24px',
    backgroundColor: errorMessage
      ? theme.colors.errorInputBackgroundColor
      : theme.colors.inputBackgroundColor
  },
  label: {
    ...theme.label,
    height: '24px',
    lineHeight: '20px',
    top: '-2px',
    fontWeight: 'bold',
    marginTop: '22px'
  }
}));

export const StyledCard = styled.div<{
  maxWidth: number;
  backgroundColor: string;
}>(props => ({
  backgroundColor: props.backgroundColor,
  width: '100%',
  maxWidth: props.maxWidth,
  boxShadow:
    '0px 1px 1.5px rgba(0, 0, 0, 0.106), 0px 0px 0.5px rgba(0, 0, 0, 0.08)',
  borderRadius: '3px'
}));
