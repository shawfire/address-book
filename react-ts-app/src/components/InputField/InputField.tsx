import * as React from 'react';
import { StyledFieldWrapper } from './styled';
import { ErrorSvg } from '../Common/ErrorSvg';
import { StyledError } from '../Common/styled';

export interface InputPropTypes {
  id: string;
  name: string;
  type: string;
  'aria-describedby'?: string;
  value: string;
  maxLength?: number;
  autoFocus?: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur: () => void;
  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  onMouseDown: (e: React.MouseEvent<HTMLInputElement, MouseEvent>) => void;
  ref: React.RefObject<HTMLInputElement>;
}

export interface InputFieldPropTypes {
  id?: string;
  inputProps: InputPropTypes;
  label: string;
  errorMessage: string;
}

function InputField({ inputProps, label, errorMessage }: InputFieldPropTypes) {
  return (
    <StyledFieldWrapper errorMessage={errorMessage}>
      <label htmlFor={inputProps.id}>{label}</label>
      <input {...inputProps} />
      {errorMessage && (
        <StyledError id={`${inputProps.id}-error`}>
          <ErrorSvg />
          &nbsp;
          {errorMessage}
        </StyledError>
      )}
    </StyledFieldWrapper>
  );
}

export default InputField;
