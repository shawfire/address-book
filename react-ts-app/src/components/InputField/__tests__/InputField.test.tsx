import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import InputField, { InputFieldPropTypes } from "../InputField";

describe("Input Field tests", () => {
  let wrapper: ReactWrapper | ShallowWrapper;
  const inputFieldProps: InputFieldPropTypes = {
    inputProps: {
      id: "testid",
      name: "testid",
      type: "text",
      value: "",
      onChange: () => {},
      onBlur: () => {},
      onKeyDown: () => {},
      onMouseDown: () => {},
      ref: React.createRef()
    },
    label: "Family name / surname",
    errorMessage: ""
  };

  test("Input Field snapshot", () => {
    wrapper = shallow(<InputField {...inputFieldProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("First Input Field snapshot", () => {
    inputFieldProps.inputProps.autoFocus = true;
    wrapper = shallow(<InputField {...inputFieldProps} />);

    expect(wrapperu).toMatchSnapshot();
  });
});
