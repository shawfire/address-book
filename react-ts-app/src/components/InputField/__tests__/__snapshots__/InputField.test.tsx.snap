// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field tests First Input Field snapshot 1`] = `ShallowWrapper {}`;

exports[`Input Field tests Input Field snapshot 1`] = `ShallowWrapper {}`;
